packages/devtools_app/lib/src/debugger/file_search.dart

```diff
@@ -114,22 +135,23 @@ List<ScriptRef> findMatches(
   List<ScriptRef> scriptRefs,
 ) {
   if (query.isEmpty) {
-    takeTopMatches(scriptRefs);
+    return scriptRefs;
   }
 
-  final exactMatches = scriptRefs
-      .where((scriptRef) => scriptRef.uri.caseInsensitiveContains(query))
-      .toList();
+  final exactMatches = [];
+  final fuzzyMatches = [];
 
-  if (exactMatches.length >= numOfMatchesToShow) {
-    return takeTopMatches(exactMatches);
+  for (final scriptRef in scriptRefs) {
+    final fullPath = scriptRef.uri;
+    final fileName = scriptRef.uri.split('/').last;
+    if (fullPath.caseInsensitiveContains(query)) {
+      exactMatches.add(scriptRef);
+    } else if (fileName.caseInsensitiveFuzzyMatch(query)) {
```

Comment:
might want to also check to see if caseInsensitiveFuzzyMatch can be optimized

Reply:
I took a look at it before - it's linear so I don't think it can be optimized further